File: problems/2231-find-first-palindromic-string-in-the-array/find-first-palindromic-string-in-the-array.js
/**
 * @param {string[]} words
 * @return {string}
 */
const firstPalindrome = (words) => {
    for (let w of words) {
        if (isPalindromic(w)) return w;
    }
    return "";
};

const isPalindromic = (w) => {
    let l = 0;
    let r = w.length - 1;
    while (l < r) {
        if (w[l] !== w[r]) return false;
        l++;
        r--;
    }
    return true;
}

File: problems/2231-find-first-palindromic-string-in-the-array/find-first-palindromic-string-in-the-array.py
class Solution:
    def isPalindome(self, w:str) -> bool:
        l = 0
        r = len(w) - 1
        while l < r:
            if w[l] != w[r]:
                return False
            l += 1
            r -= 1
        return True
        

    def firstPalindrome(self, words: List[str]) -> str:
        for w in words:
            if self.isPalindome(w):
                return w
        return ""

